be/src/vec/exec/vjson_scanner.cpp

```diff
@@ -0,0 +1,601 @@
+    return Status::OK();
+}
+
+// for simple format json
+// set valid to true and return OK if succeed.
+// set valid to false and return OK if we met an invalid row.
+// return other status if encounter other problmes.
+Status VJsonReader::_set_column_value(rapidjson::Value& objectValue, std::vector<MutableColumnPtr>& columns,
+                                    const std::vector<SlotDescriptor*>& slot_descs, bool* valid) {
+    if (!objectValue.IsObject()) {
+        // Here we expect the incoming `objectValue` to be a Json Object, such as {"key" : "value"},
+        // not other type of Json format.
+        std::string err_msg("Expect json object value");
+        RETURN_IF_ERROR(_append_error_msg(objectValue, err_msg, valid));
+        return Status::OK();
+    }
+
+    int nullcount = 0;
+    int ctx_idx = 0;
+    for (auto slot_desc : slot_descs) {
+        int dest_index = ctx_idx++;
+        auto* column_ptr = columns[dest_index].get();
+        rapidjson::Value::ConstMemberIterator it = objectValue.MemberEnd();
+
+        if (_fuzzy_parse) {
+            auto idx_it = _name_map.find(slot_desc->col_name());
+            if (idx_it != _name_map.end() && idx_it->second < objectValue.MemberCount()) {
+                it = objectValue.MemberBegin() + idx_it->second;
+            }
+        } else {
+            it = objectValue.FindMember(
+                    rapidjson::Value(slot_desc->col_name().c_str(), slot_desc->col_name().size()));
+        }
+
+        if (it != objectValue.MemberEnd()) {
+            const rapidjson::Value& value = it->value;
+            RETURN_IF_ERROR(_write_data_to_column(&value, slot_desc, column_ptr, valid));
+            if (!(*valid)) {
+                return Status::OK();
+            }
+        } else { // not found
+            if (slot_desc->is_nullable()) {
+                auto* nullable_column = reinterpret_cast<vectorized::ColumnNullable*>(column_ptr);
+                nullable_column->insert_data(nullptr, 0);
+                nullcount++;
+            } else {
+                fmt::memory_buffer error_msg;
+                fmt::format_to(error_msg, "The column `{}` is not nullable, but it's not found in jsondata.", slot_desc->col_name());
+                std::string err_msg = fmt::to_string(error_msg);
+                RETURN_IF_ERROR(_append_error_msg(objectValue, err_msg, valid));
+                break;
+            }
+        }
+    }
+
+    if (nullcount == slot_descs.size()) {
+        std::string err_msg("All fields is null, this is a invalid row.");
+        RETURN_IF_ERROR(_append_error_msg(objectValue, err_msg, valid));
+        return Status::OK();
+    }
+    *valid = true;
+    return Status::OK();
+}
+
+Status VJsonReader::_write_data_to_column(rapidjson::Value::ConstValueIterator value, SlotDescriptor* slot_desc,
+                                        vectorized::IColumn* column_ptr, bool* valid) {
+    const char* str_value = nullptr;
+    uint8_t tmp_buf[128] = {0};
```

Comment:
Why not just use `char` directly?

Reply:
Is 128 too big here?

Reply:
yes,  uint8_t is same as char, I will re-check.

Reply:
> Is 128 too big here?
more is better than less ?